### PR TITLE
Change setup to being based on setup.cfg, rather than setup.py

### PR DESCRIPTION
**Pull request by Clockwork-Muse**
Setup.py is retained to allow for editable installs, but is otherwise no longer required.

### CODE REVIEW

**Comment by Moguri** [comment]
What is the benefit of this change? It seems to make some stuff more complicated, and I'm not sure what it improves.

**Comment by Clockwork-Muse** [comment]
nnn.... it's mostly look-ahead, because the trend in python is to deprecate `setup.py` in favor of `setup.cfg`.
The testing-related stuff was done because apparently `setup-requires` and `testing-requires` can bypass some pip security checks, and `pytest-runner` is being deprecated.

**Comment by Clockwork-Muse** [comment]
I didn't see that there was a `travis.yml` file - I was looking for the `.github` directory.  If you do go through with that I'd need to adjust the script there too.

File: setup.cfg [hunk 1]
```diff
@@ -18,13 +19,30 @@ classifiers =
...
+    panda3d-simplepbr >= 0.6
+    importlib-metadata >= 1.0 ; python_version < "3.8"
+
+[options.entry_points]
+console_scripts =
+    gltf2bam = gltf.cli:main
+gui_scripts =
+    gltf-viewer = gltf.viewer:main
+    octodns-report = octodns.cmds.report:main
+    octodns-sync = octodns.cmds.sync:main
```
Comment (rdb) [inline setup.cfg:36]: OctoDNS?

File: setup.cfg [hunk 2]
```diff
@@ -1,13 +1,14 @@
 [metadata]
-name = panda3d-gltf
+name = panda3d-glTF
```
Comment (rdb) [inline setup.cfg:2]: Name change?  Most packages in the Python ecosystem seem to be lowercase-named.

File: gltf/cli.py [hunk 3]
```diff
@@ -15,7 +21,7 @@ def main():
     parser.add_argument(
         '--version',
         action='version',
-        version='%(prog)s {}'.format(__version__)
+        version='%(prog)s {}'.format(metadata.version('panda3d-gltf'))
```
Comment (rdb) [inline gltf/cli.py:24]: This seems like the kind of thing that causes issues when using a deployment system, has this been tested to work in a deployed environment?

File: setup.cfg (hunk 2 above)
Comment (rdb) [reply setup.cfg:2]: It is, but by that logic shouldn't it be Panda3D-glTF, which looks even weirder as a package name?

File: gltf/cli.py (hunk 3 above)
Comment (Clockwork-Muse) [reply gltf/cli.py:24]: I tested it by building it (in both 3.6 and 3.8), installing the wheel in a clean venv, and calling `gltf2bam --version`.  Did you mean a deployed, built, panda application?  Something else?

File: setup.cfg (hunk 1 above)
Comment (Clockwork-Muse) [reply setup.cfg:36]: copy-paste error from the example I found

File: setup.cfg (hunk 2 above)
Comment (Clockwork-Muse) [reply setup.cfg:2]: Ah, the remark was just an aside for another reason why I didn't change it initially.  I personally don't really care.

**Comment by Moguri** [comment]
Thanks for your efforts on this. I went ahead updated this repo to match my other projects that have ditched setup.py.

**Pull request closed without merging**